app/main/views/frameworks.py

```diff
@@ -260,6 +260,26 @@ def framework_submission_services(framework_slug, lot_slug):
     if framework['status'] == 'pending' and declaration_status != 'complete':
         abort(404)
 
+    try:
+        previous_framework_slug = content_loader.get_metadata(framework['slug'], 'copy_services', 'source_framework')
+    except ContentNotFoundError:
+        previous_framework_slug = None
+        previous_services = []
+        previous_framework = {}
+
+    if previous_framework_slug:
+        previous_framework = get_framework_or_404(data_api_client, previous_framework_slug)
+        previous_services = data_api_client.find_services(
+            supplier_id=current_user.supplier_id,
+            framework=previous_framework_slug,
+            lot=lot_slug,
+            status='published',
+        )["services"]
+
+    previous_services_still_to_copy = len([
+        service for service in previous_services if not service['copiedToFollowingFramework']
```

Comment:
A `filter` here would be slightly more Pythonic, I think. But no big deal either way.

`previous_services_still_to_copy = len(filter(lambda x: x['copiedToFollowingFramework'] is False, services))`

Reply:
I'm gonna keep as it actually. `filter` objects don't have `len()`, so it becomes `len(list(filter(....)))` which is a bit naf.

Reply:
🤦‍♂️ 

I thought it would just iterate over the filtered list. Sometimes Python 3 feels a bit like it's made things a bit weird.

Reply:
Yeah, i find `filter` and `map` objects really unintuitive.